### PR TITLE
Support spawn for auth

### PR DESCRIPTION
**Pull request by joii2020**
* Support spawn for auth.
* Add testcase for spawn (not yet added to CI).
* Update Cargo.toml for root project(Old version of openssl reported an error).
* Replace openssl to mbedtls in auth_rust.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -39,7 +39,7 @@ const BINARIES: &[(&str, &str)] = &[
     ),
     (
         "omni_lock",
-        "635f78eba450cb2f73f113022ff62e4bbfb5a39b7368c375c6a731ba4c85c59e"
```
Comment (XuJiandong) [inline build.rs:42]: We can't update this. Revert related changes.

File: Makefile [hunk 2]
```diff
@@ -9,7 +9,7 @@ SECP256K1_SRC_20210801 := deps/secp256k1-20210801/src/ecmult_static_pre_context.
 
 XUDT_RCE_CFLAGS=$(subst ckb-c-std-lib,ckb-c-stdlib-20210713,$(CFLAGS)) -I deps/sparse-merkle-tree/c
 
-OMNI_LOCK_CFLAGS :=$(subst ckb-c-std-lib,ckb-c-stdlib-20210801,$(CFLAGS)) -I deps/sparse-merkle-tree/c
+OMNI_LOCK_CFLAGS :=$(subst ckb-c-std-lib,ckb-c-stdlib-2023,$(CFLAGS)) -I deps/sparse-merkle-tree/c
```
Comment (XuJiandong) [inline Makefile:12]: Don't update this.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: Cargo.lock [hunk 3]
```diff
@@ -1,27 +1,47 @@
 # This file is automatically @generated by Cargo.
 # It is not intended for manual editing.
+version = 3
```
Comment (XuJiandong) [inline Cargo.lock:3]: Try not to update cargo.lock since it's not necessary?

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: Cargo.lock (hunk 3 above)
Comment (joii2020) [reply Cargo.lock:3]: This project depend ckb-script, the ckb-script depend ckb-log, the ckb-log depend sentry, the sentry depend reqwest, the reqwest is a higher-level HTTP client.

**Pull request merged**